core/arch/arm/mm/tee_pager.c

```diff
@@ -1066,6 +1069,8 @@ static struct tee_pager_pmem *tee_pager_get_page(struct tee_pager_area *area)
 		area_set_entry(pmem->area, pmem->pgidx, 0, 0);
 		pgt_dec_used_entries(pmem->area->pgt);
 		tlbi_mva_allasid(area_idx2va(pmem->area, pmem->pgidx));
+		if (a & (TEE_MATTR_PX | TEE_MATTR_UX))
+			cache_op_inner(ICACHE_INVALIDATE, NULL, 0);
```

Comment:
You're right, since aliased pages never are executable they can never end up in the icache either and other than that there's no aliasing of the pages. I'll update this pull request accordingly.